### PR TITLE
Use color constants

### PR DESCRIPTION
**Pull request by PeterDRS**
Use (new) color constants with new values

### CODE REVIEW

File: img/signals/signal.php [hunk 1]
```diff
@@ -250,13 +252,14 @@
...
 	<!ENTITY background "fill:#000000;">	
 	<!ENTITY white "fill:#FFFFFF;">	
-	<!ENTITY gray "fill:#444444;">	
-	<!ENTITY red "fill:red;">	
-	<!ENTITY yellow "fill:yellow;">	
-	<!ENTITY green "fill:#38FFF5;">
-		
+	<!ENTITY gray "fill:#444444;">
+<?php
+	echo("<!ENTITY red "fill:" . $color_red . ";\">");
```
Comment (sb12) [inline img/signals/signal.php]: color_red is defined as a constant and is not a variable.

File: img/signals/signal.php [hunk 2]
```diff
@@ -250,13 +252,14 @@
...
 	<!ENTITY white "fill:#FFFFFF;">	
-	<!ENTITY gray "fill:#444444;">	
-	<!ENTITY red "fill:red;">	
-	<!ENTITY yellow "fill:yellow;">	
-	<!ENTITY green "fill:#38FFF5;">
-		
+	<!ENTITY gray "fill:#444444;">
+<?php
+	echo("<!ENTITY red "fill:" . $color_red . ";\">");	
+	echo("<!ENTITY yellow "fill:" . $color_yellow . ";\">");
```
Comment (sb12) [inline img/signals/signal.php]: This should be `echo('<!ENTITY yellow "fill:' . $color_yellow . ';\">');`
You also do not need the brackets.

Comment (PeterDRS) [inline img/signals/signal.php]: It's very hard to write code without having a possibility to test it.

**Comment by sb12** [comment]
Why having an extra file to define the colors?
Ok they might be hard to find in line 250, but why not just have the constants defined in the first rows?

**Comment by PeterDRS** [comment]
> Why having an extra file to define the colors?

Because the colors should ABSOLUTELY have NOTHING to do with any code. Not even the svg code.

Someone changes the colors: Look at file colors.php. Change it. Everybody sees directly with the pull request: Nothing functional, only asthetics.

You change the signals.php. You split it up. Who knows. The colors stay safe in colors.php. No longer searching for it in any files. ;)

**Comment by PeterDRS** [comment]
Very good. From now on I know where to look to change the colors, if neccessary again.

**Pull request merged**